### ISSUE TITLE
aws-cdk-lib and constructs get fixed version in devDependencies with AwsCdkConstructLibrary

### ISSUE THREAD

**Issue by Hi-Fi** [projen/projen#1409]
When generating aws cdk construct project with Projen, I get peerDependencies right with caret.

Issue is, that devDependencies get only exact versions, which makes it quite hard to utilize construct during the development as directory import as it requires exactly same versions of dependencies to work.

For aws-cdk-lib issue is not so bit (as that's updated often), but it feels a bit annoying to have to update also constructs version manually all the time.

Is there reason for strict versions?

Strict versions created in:
- https://github.com/projen/projen/blob/eb1283fbadd13d2d25fe977dbecabddab523bda8/src/awscdk/awscdk-deps.ts#L237
- https://github.com/projen/projen/blob/eb1283fbadd13d2d25fe977dbecabddab523bda8/src/awscdk/awscdk-deps.ts#L211

**Comment by chrisjgray** [comment]
I also have the same issue. When I do local debugging to test out new features of my library, I have to hardcode my app to use the exact same versions of the cdk library and constructs.

**Comment by Hi-Fi** [comment]
Seems that code is changed a bit, but still issue seems to be present.

**Comment by Hi-Fi** [comment]
Actually issue seems to be at https://github.com/projen/projen/blob/main/src/javascript/node-package.ts#L904-L913.

That clears version to exact semver, causing dev dependencies to have different version from peer dependencies

**Comment by Hi-Fi** [comment]
@eladb  At the https://github.com/projen/projen/blob/main/test/node-project.test.ts#L103 there's comment that devDependencies are always added as pinned versions, which makes it hard to develop construct by including it app in file system (due to mismatch in versions).

Is there some reason for that approach that peer and dev dependencies are put different way? It would be easy to remove the part linked in previous one, but that's change in the current logic.

**Comment by eladb** [comment]
Yes, it's intentional. The rationale is to test against the minimum supported compatible 
 version.

If you say that you support version 1.2.3 and above (`^1.2.3`) then you should not rely on features that only exist in 1.3.0. 

Semantic versioning ensures that if your consumers install a compatible later version then your code will work.

So a recommended practice is to install the minimum supported version when you develop and test so you won't accidentally take a dependency on something your consumers won't have.

If you wish to use a new feature, you should bump your peer requirement.

**Comment by eladb** [comment]
As for local linking, you are right. The experience is broken, but we should address this devX need instead of breaking compatibility.

Can you raise an issue about local linking and we can discuss some options in there?
